### DISCUSSION TITLE
Lesson 12: DeFI Stablecoin : The Math behind heath factor is looking a bit confusing

### DISCUSSION THREAD

**Discussion by AtulBlockDev** [Cyfrin/foundry-full-course-cu#368] [Q&A] [answered]
I don't know why but I am not getting the MATH implemented in the logic for HealthFactor. 
 
The _healthFactor function is 
(collateralAdjustedForThreshold * PRECISION)/totalDscMinted 

The collateralAdjustedForThreshold is 

collateralValueInUsd * LIQUIDATION_THRESHOLD/100

The CollateralValueInUsd must be already in e18, because the code is 

priceFeedPrice * 1e10 * amount (in e18)/1e18

Let's cancel the e18 of amount with e18 in division. 

And the collateralValue in USD will be {something*e18}. Right?

Now suppose if we mint 100dsc tokens then we get,

The Health Factor will be

AdjustedCollateralValueInUsd (in e18) * PRECISION (e18) / 100

That will be {something×e36)/100

So the helath factor will be in terms of e36 with 36 decimals. 

I am not getting this MATH at all. What am I missing?

**Comment by asobiesk** [comment, marked as answer]
Your math is mostly correct. However, the final line of _healthFactor method is:

```solidity
return (collateralAdjustedForTreshhold * PRECISION) / totalDscMinted;
```
collateralAdjustedForTreshhold is indeed in 1e18, just how you calculated. PRECISION is 1e18, so is the totalDscMinted. 

(collateralAdjustedForTreshhold<sup>[1e18]</sup> * 1e18) / totalDscMinted<sup>[1e18]</sup>

So the result will be in 1e18 as well:

(collateralAdjustedForTreshhold * 10<sup>18</sup> * 10<sup>18</sup>) / (totalDscMinted * 10<sup>18</sup>) = (collateralAdjustedForTreshhold/totalDscMinted)*10<sup>18</sup> = healthFactor<sup>[1e18]</sup>

Your mistake was here:

> AdjustedCollateralValueInUsd (in e18) * PRECISION (e18) / 100

 > That will be {something×e36)/100

 > So the helath factor will be in terms of e36 with 36 decimals.

At this point we do not divide by 100, but by **totalDscMinted**, which has 18 decimals. Therefore extra 36 decimals / extra 18 decimals = extra 18 decimals. Health factor will therefore be in terms of **1e18**.

It's worth mentioning that just after that Patrick says that this health factor should be greater than MIN_HEALTH_FACTOR, which he initializes to 1. This is wrong, because our health factor has 18 extra decimals. MIN_HEALTH_FACTOR should therefore be 1e18. It is corrected later though.
